recipe/meta.yaml

```diff
@@ -1,78 +1,72 @@
-    - mv lib/11/*.so* $PREFIX/lib/                                         # [linux and cuda_compiler_version == "11.1"]
+    - mv lib/{{ cuda_major }}/*.so* $PREFIX/lib/                           # [linux]
     - patchelf --add-needed libcudart.so $PREFIX/lib/libcutensor.so        # [ppc64le]
 
     - copy include\\cutensor.h %LIBRARY_INC%\\                             # [win64]
     - copy include\\cutensorMg.h %LIBRARY_INC%\\                           # [win64]
     - mkdir %LIBRARY_INC%\\cutensor                                        # [win64]
     - copy include\\cutensor\\types.h %LIBRARY_INC%\\cutensor              # [win64]
-    - del lib\\{{ cuda_compiler_version }}\\*static*                       # [win64 and cuda_compiler_version in ("10.2", "11.0")]
-    - copy lib\\{{ cuda_compiler_version }}\\*.dll %LIBRARY_BIN%\\         # [win64 and cuda_compiler_version in ("10.2", "11.0")]
-    - copy lib\\{{ cuda_compiler_version }}\\*.lib %LIBRARY_LIB%\\         # [win64 and cuda_compiler_version in ("10.2", "11.0")]
-    - del lib\\11\\*static*                                                # [win64 and cuda_compiler_version in ("11.1", )]
-    - copy lib\\11\\*.dll %LIBRARY_BIN%\\                                  # [win64 and cuda_compiler_version in ("11.1", )]
-    - copy lib\\11\\*.lib %LIBRARY_LIB%\\                                  # [win64 and cuda_compiler_version in ("11.1", )]
-  ignore_run_exports:
-    - cudatoolkit
+    - del lib\\{{ cuda_major }}\\*static*                                  # [win64]
+    - copy lib\\{{ cuda_major }}\\*.dll %LIBRARY_BIN%\\                    # [win64]
+    - copy lib\\{{ cuda_major }}\\*.lib %LIBRARY_LIB%\\                    # [win64]
   run_exports:
     - {{ pin_subpackage('cutensor') }}
   missing_dso_whitelist:
-    # suppress warning, as these are included in the run dependency
-    - '*/libcublasLt.so*'  # [linux and cuda_compiler_version in ("11.0", "11.1")]
-    - '*/cublasLt64*.dll'  # [win64 and cuda_compiler_version in ("11.0", "11.1")]
+    - "*/libcuda.so.*"    # [linux]
+    - "*/nvcuda.dll"      # [win]
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - {{ compiler('cxx') }}
+    - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    - sysroot_{{ target_platform }} 2.17    # [linux64]
   host:
-    - patchelf >=0.12  # [linux]
+    #- patchelf >=0.12  # [linux]
+    - cuda-version {{ cuda_major }}.0
+    - libcublas
   run:
-    - cudatoolkit {{ cuda_compiler_version }}  # [cuda_compiler_version in ("10.2", "11.0")]
-    - cudatoolkit >=11.1,<12                   # [cuda_compiler_version == "11.1"]
+    - {{ pin_compatible("cuda-version", max_pin="x") }}
+    - libcublas
   run_constrained:
     # Only GLIBC_2.17 or older symbols present
     - __glibc >=2.17      # [linux]
```

Comment:
Is this constraint necessary? I think `__glibc` [is a strong run-export of `sysroot`](https://github.com/conda-forge/linux-sysroot-feedstock/blob/dd0fee5e8aca60f112dc872a1caac9ab0a0b5cda/recipe/meta.yaml#L117) which should enforce this constraint (though I'm not totally sure about the cross-target logic in that recipe). A related topic came up in the CUDA 12 recipe for libcublas and we removed it: https://github.com/conda-forge/staged-recipes/pull/21901/commits/1606772f4962578f612d6f6fee9d8cfee0238cc5

Reply:
It's a good question. This section has been there since ~3 yrs ago, and I bet we copied it from elsewhere. Things might be different now. Let me keep this thread open.